modules/distribution/index.js

```diff
@@ -4,46 +4,48 @@ const process = require("process");
 const fs = require("fs");
 const log = require("log");
 const {spawn} = require("child_process");
-const {createWindow} = require("./src/app")
+const {createWindow} = require("./src/app");
+const {createErrorWindow} = require('./src/error-window');
+const {ErrorCodes} = require('./src/error-codes');
 
 // Keep a global reference of the window object, if you don't, the window will
 // be closed automatically when the JavaScript object is garbage collected.
 let win,
     serviceProcess,
     logger = new log("info"),
-    logsDir = path.join(__dirname, "logs"),
     appDir = app.getAppPath(),
+    logsDir = path.join(appDir, "..", "..", "logs"),
     ballerinaHome = path.join(__dirname, "bre");
 
 function createLogger(){
 	if (!fs.existsSync(logsDir)){
 		fs.mkdirSync(logsDir);
 	}
-	fs.access(logsDir, fs.W_OK, function(err) {
-		if(err){
-			logger.error("can't write to log folder.");
-		}
-		else{
-			logger = new log("info", fs.createWriteStream(path.join(logsDir, "app.log")));
-		}
-	});
```

Comment:
indentation here uses tabs no? shall we stick to spaces? otherwise it kind of messes up indentation.